pkg/cmd/dev/testlogic.go

```diff
@@ -297,6 +305,39 @@ func (d *dev) testlogic(cmd *cobra.Command, commandLine []string) error {
 	return nil
 }
 
+// This function determines if any test_logic or execbuilder/testdata files were
+// modified in the current branch, and if so, determines if we should re-generate logic tests.
+func (d *dev) shouldGen(ctx context.Context) (bool, error) {
```

Comment:
Let's call this `shouldGenerateLogicTests` or something -- `shouldGen` is not descriptive enough.